src/Adapter/Language/LanguageImageManager.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Language;
+
+use ImageType;
+
+/**
+ * Handles language images (flag, "no image" placeholders)
+ */
+class LanguageImageManager
+{
+    /**
+     * Path where images are saved to
+     */
+    const IMG_PATH = _PS_ROOT_DIR_ . '/img/l/';
+
+    /**
+     * Path where flags are stored
+     */
+    const FLAGS_SOURCE = _PS_BO_ALL_THEMES_DIR_ . 'new-theme/static/flags/%s.jpg';
+
+    const FALLBACK_FLAG_SOURCE = self::IMG_PATH . 'none.jpg';
+
+    const IMAGE_DIRECTORIES = [_PS_CAT_IMG_DIR_, _PS_MANU_IMG_DIR_, _PS_PROD_IMG_DIR_, _PS_SUPP_IMG_DIR_];
+
+    /**
+     * Sets up the language flag image for the given language
+     *
+     * @param string $localeCode IETF language tag
+     * @param int $langId Language id
+     * @param string|null $flagCode If provided, use this flag code. By default, auto-detect using locale code.
+     *
+     * @throws \Exception
+     */
+    public function setupLanguageFlag(string $localeCode, int $langId, ?string $flagCode = null): void
+    {
+        $flagCode = $flagCode ?? $this->getFlagCountryCodeFromLocale($localeCode);
+
+        $flagPath = $this->getFlagPath($flagCode);
+
+        if (!file_exists($flagPath)) {
+            $flagPath = self::FALLBACK_FLAG_SOURCE;
```

Comment:
Same as previous about final class
```suggestion
            $flagPath = static::FALLBACK_FLAG_SOURCE;
```

Reply:
Done